src/api/stores.py

```diff
@@ -36,138 +36,129 @@ def get_stores():
+        
+        catalog = conn.execute(fetch_catalog, {"store_id": store_id})
+
+    return_list = [
+        {
+        "food_id": item.food_id,
+        "item": item.name,
+        "quantity": item.quantity,
+        "price": f"${item.price / 100:.2f}"
+        }
+        for item in catalog
+        
+    ]
+    
+    return return_list
 
-class PricesRequest(BaseModel):
-    food_id: int = Field(ge=0)
-    max_price: Optional[int] = Field(default=0, ge=0)
-    max_stores: Optional[int] = Field(default=3, ge=1)
 
 @router.post("/compare-prices")
-def compare_prices(request: PricesRequest):
+def compare_prices(food_id: int, 
+            max_stores: int = Query(3, description="How many stores would you like to see", gt=0)
+                   ):
     """
     Find the stores with the best prices
     """
-
-    query = """
-    SELECT store.store_id AS id, store.name AS name, price
-    FROM store
-    JOIN catalog ON catalog.store_id = store.store_id
-    JOIN catalog_item ON catalog_item.catalog_id = catalog.catalog_id
-    WHERE catalog_item.food_id = :food_id
-    """
-    if request.max_price > 0:
-        query += "AND catalog_item.price <= :max_price"
-
-    query += """
-    ORDER BY price ASC
-    LIMIT :max_stores
-    """
-        
+    find_stores = sqlalchemy.text("""
+        SELECT store.store_id AS id, store.name AS store, food_item.name, price,
+                RANK() OVER (PARTITION BY catalog_item.food_id ORDER BY price) AS rank
+        FROM store
+        JOIN catalog ON catalog.store_id = store.store_id
+        JOIN catalog_item ON catalog_item.catalog_id = catalog.catalog_id
+        JOIN food_item ON food_item.food_id = catalog_item.food_id
+        WHERE catalog_item.food_id = :food_id
+        ORDER BY price ASC
+        LIMIT :max_stores
+    """)
+    
+    
+    print(type(food_id), type(max_stores))
     query_params = [{
-        "food_id": request.food_id,
-        "max_price": request.max_price,
-        "max_stores": request.max_stores
+        "food_id": food_id,
+        "max_stores": max_stores
     }]
 
-    try:
-        with db.engine.connect().execution_options(isolation_level="REPEATABLE READ") as conn:
-            with conn.begin():
-                result = conn.execute(sqlalchemy.text(query), query_params)
-
-        result = result.fetchall()
-
-        if len(result) <= 0:
-            raise NoResultFound
-
-    except NoResultFound:
-        raise HTTPException(
-            status_code=status.HTTP_404_NOT_FOUND,
-            detail="No matching results"
-        )
-    except IntegrityError as e:
-        return e
-
-    return_object = []
-    place = 1
-    for line in result:
-        return_object.append({
-            "store_id": line.id,
-            "store_name": line.name,
-            "price": line.price,
-            "rank": place
-        })
-        place += 1
+    with db.engine.connect().execution_options(isolation_level="REPEATABLE READ") as conn:
+        with conn.begin():
```

Comment:
I don't think the second with statement is necessary here. I deleted it in testing and it still worked, and still threw the error when it was supposed to. What were you thinking the purpose of it was?

Reply:
Im unsure that part wasn't me